contrib/scalar/scalar.c

```diff
@@ -141,14 +141,56 @@ static int set_config(const char *file, const char *fmt, ...)
 	return res;
 }
 
-static char *remote_default_branch(const char *dir)
+static int retrieve_cache_server_url(const char *dir, const char *url,
+				     char **cache_server_url)
+{
+	struct child_process cp = CHILD_PROCESS_INIT;
+	struct strbuf out = STRBUF_INIT;
+	const char *needle = "\"CacheServers\":[";
+	const char *url_needle = "\"Url\":\"";
+	const char *global_default_needle = "\"GlobalDefault\":true";
+	const char *p, *curly, *dq;
+	char *res = NULL;
+
+	cp.git_cmd = 1;
+	cp.dir = dir; /* gvfs-helper requires a Git repository */
+	strvec_pushl(&cp.args, "gvfs-helper", "--remote", url, "config", NULL);
+	if (pipe_command(&cp, NULL, 0, &out, 512, NULL, 0)) {
+		strbuf_release(&out);
+		return -1; /* error out quietly */
+	}
+
+	p = strstr(out.buf, needle);
+	if (p)
+		p += strlen(needle);
+	while (*p == '{') {
+		curly = strchrnul(p, '}');
```

Comment:
Since the GVFS protocol design is essentially frozen, I think we will be able to rely on the `CacheServers` attribute being a shallow object. Read: I don't think we need to care about stacking.

Reply:
Right. This works.